service/api.go

```diff
@@ -230,61 +230,12 @@ func (as *APIServer) validateProjectConfig(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
-	errs := validator.ValidationErrors{}
-	var projectRef *model.ProjectRef
-	if input.ProjectID != "" {
-		projectRef, err = model.FindMergedProjectRef(input.ProjectID, "", false)
-		if err != nil {
-			validationErr = validator.ValidationError{
-				Message: "error finding project; validation will proceed without checking project settings",
-				Level:   validator.Warning,
-			}
-			errs = append(errs, validationErr)
-		} else if projectRef == nil {
-			validationErr = validator.ValidationError{
-				Message: "project does not exist; validation will proceed without checking project settings",
-				Level:   validator.Warning,
-			}
-			errs = append(errs, validationErr)
-		} else {
-			isConfigDefined := projectConfig != nil
-			errs = append(errs, validator.CheckProjectSettings(ctx, evergreen.GetEnvironment().Settings(), project, projectRef, isConfigDefined)...)
-		}
-	} else {
-		validationErr = validator.ValidationError{
-			Message: "no project specified; validation will proceed without checking project settings",
-			Level:   validator.Warning,
-		}
-		errs = append(errs, validationErr)
-	}
-
-	errs = append(errs, validator.CheckProjectErrors(r.Context(), project, input.IncludeLong)...)
-	if projectConfig != nil {
-		errs = append(errs, validator.CheckProjectConfigErrors(projectConfig)...)
-	}
+	projectRef, err := model.FindMergedProjectRef(input.ProjectID, "", false)
```

Comment:
I pass the error in to the check project function because it handles the output relating to the error. I updated the implementation with a better flow and different data types.

Reply:
Oh yeah I forgot to update this comment. Since its unusual not to check the error here, we should add a comment, or else someone else is gonna assume this is a mistake and change it. Ideally we'd also name it something less generic than err since we're using it as a variable, like "projectRefErr"